src/python/pants/backend/python/packaging/pyoxidizer/rules.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import logging
+from dataclasses import dataclass
+
+from pants.backend.python.packaging.pyoxidizer.config import PyOxidizerConfig
+from pants.backend.python.packaging.pyoxidizer.subsystem import PyOxidizer
+from pants.backend.python.packaging.pyoxidizer.target_types import (
+    PyOxidizerConfigSourceField,
+    PyOxidizerDependenciesField,
+    PyOxidizerEntryPointField,
+    PyOxidizerUnclassifiedResources,
+)
+from pants.backend.python.util_rules.pex import Pex, PexProcess, PexRequest
+from pants.core.goals.package import BuiltPackage, BuiltPackageArtifact, PackageFieldSet
+from pants.core.util_rules.source_files import SourceFiles, SourceFilesRequest
+from pants.engine.fs import (
+    CreateDigest,
+    Digest,
+    DigestContents,
+    FileContent,
+    MergeDigests,
+    Snapshot,
+)
+from pants.engine.process import ProcessResult
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.engine.target import (
+    DependenciesRequest,
+    FieldSetsPerTarget,
+    FieldSetsPerTargetRequest,
+    Targets,
+)
+from pants.engine.unions import UnionRule
+from pants.util.logging import LogLevel
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class PyOxidizerFieldSet(PackageFieldSet):
+    required_fields = (PyOxidizerDependenciesField,)
+
+    entry_point: PyOxidizerEntryPointField
+    dependencies: PyOxidizerDependenciesField
+    unclassified_resources: PyOxidizerUnclassifiedResources
+    template: PyOxidizerConfigSourceField
+
+
+@rule(level=LogLevel.DEBUG)
+async def package_pyoxidizer_binary(
+    pyoxidizer: PyOxidizer, field_set: PyOxidizerFieldSet
+) -> BuiltPackage:
+    logger.info(f"Incoming package_pyoxidizer_binary field set: {field_set}")
+    targets = await Get(Targets, DependenciesRequest(field_set.dependencies))
+    target = targets[0]
+
+    logger.info(
+        f"Received these targets inside pyox targets: {target.address.target_name}"
+    )
+
+    packages = await Get(
+        FieldSetsPerTarget,
+        FieldSetsPerTargetRequest(PackageFieldSet, [target]),
+    )
+    logger.info(f"Retrieved the following FieldSetsPerTarget {packages}")
+
+    built_packages = await MultiGet(
+        Get(BuiltPackage, PackageFieldSet, field_set)
+        for field_set in packages.field_sets
+    )
+
+    # TODO: Can this be walrus'd? Double for with repeated artifact.relpath is ugly
```

Comment:
Pants currently supports running under versions which don't have the walrus operator, so: not yet.

But this PR is a useful step toward resolving that! =)